integration-tests/bats/sql-remotes-server.bats

```diff
@@ -0,0 +1,556 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+remotesrv_pid=
+setup() {
+    setup_common
+    cd $BATS_TMPDIR
+    mkdir remotes-$$
+    mkdir remotes-$$/empty
+    echo remotesrv log available here $BATS_TMPDIR/remotes-$$/remotesrv.log
+    remotesrv --http-port 1234 --dir ./remotes-$$ &> ./remotes-$$/remotesrv.log 3>&- &
+    remotesrv_pid=$!
+    cd dolt-repo-$$
+    mkdir "dolt-repo-clones"
+}
+
+teardown() {
+    teardown_common
+    kill $remotesrv_pid
+    rm -rf $BATS_TMPDIR/remotes-$$
+}
+
+@test "sql-remotes-server: push with only one argument" {
+    skip "todo implement SQL push, pull, fetch"
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    run dolt sql -q "select dolt_push('test-remote')"
+    [ "$status" -eq 1 ]
+}
+
+@test "sql-remotes-server: push and pull master branch from a remote" {
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    # TODO dolt_push
+    #run dolt sql -q "select dolt_push('test-remote', 'master')"
+    run dolt push test-remote master
+    [ "$status" -eq 0 ]
+    [ -d "$BATS_TMPDIR/remotes-$$/test-org/test-repo" ]
+    run dolt sql -q "select dolt_pull('test-remote')"
+    [ "$status" -eq 0 ]
+    #[[ "$output" =~ "Everything up-to-date" ]] || false
+}
+
+@test "sql-remotes-server: push and pull non-master branch from remote" {
+    skip "todo implement SQL push, pull, fetch"
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    dolt checkout -b test-branch
+    # TODO dolt_push
+    run dolt sql -q "select dolt_push('test-remote', 'test-branch')"
+    [ "$status" -eq 0 ]
+    run dolt sql -q "select dolt_pull('test-remote')"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Everything up-to-date" ]] || false
+}
+
+@test "sql-remotes-server: push and pull from non-master branch and use --set-upstream" {
+    skip "todo implement SQL push, pull, fetch"
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    dolt checkout -b test-branch
+    run dolt sql -q "select dolt_push('--set-upstream', 'test-remote', 'test-branch')"
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ "panic:" ]] || false
+    dolt sql -q "create table test (pk int, c1 int, primary key(pk))"
+    dolt add .
+    dolt commit -m "Added test table"
+    run dolt sql -q "select dolt_push()"
+    [ "$status" -eq 0 ]
+}
+
+@test "sql-remotes-server: push and pull with docs from remote" {
+    skip "todo pull docs?"
```

Comment:
We don't touch the filesystem when running the SQL server. A SQL session doesn't really have a notion of files the way the CLI does.